api/src/main/java/org/openmrs/module/patientqueueing/api/impl/PatientQueueingServiceImpl.java

```diff
@@ -147,4 +150,29 @@ public String generateVisitNumber(Location location, Patient patient) {
         return dateString + "-" + locationName + "-" + zeroesToAppend + nextNumberInQueue;
     }
 
+    /**
+     * @see org.openmrs.module.patientqueueing.api.PatientQueueingService#getPatientQueueListBySearchParams(java.lang.String, java.util.Date, java.util.Date, org.openmrs.Location, org.openmrs.Location, org.openmrs.module.patientqueueing.model.PatientQueue.Status)
+     */
+    @Override
+    public List<PatientQueue> getPatientQueueListBySearchParams(String searchString, Date fromDate, Date toDate, Location locationTo, Location locationFrom, PatientQueue.Status status) {
+        if (fromDate == null || toDate == null) {
+            fromDate = OpenmrsUtil.firstSecondOfDay(new Date());
```

Comment:
What happens when only one is null but the other have a value?
Secondly, why do you assign a value when the caller passed in null?